### ISSUE TITLE
I can't install any modules

### ISSUE THREAD

**Issue by burdokz** [nodeschool/discussions#947]
Hi, I just got started using node and I think I made the setups correctly.

The node --version works (v0.12.0) so I think it is correctly installed but whenever I try to install something else like *npm install figlet* it just returns a lot of error logs.

```
0 info it worked if it ends with ok
1 verbose cli [ 'D:\\nodejs\\\\node.exe',
1 verbose cli   'D:\\nodejs\\node_modules\\npm\\bin\\npm-cli.js',
1 verbose cli   'install',
1 verbose cli   'figlet' ]
2 info using npm@1.4.9
3 info using node@v0.12.0
4 verbose node symlink D:\nodejs\\node.exe
5 verbose cache add [ 'figlet', null ]
6 verbose cache add name=undefined spec="figlet" args=["figlet",null]
7 verbose parsed url { protocol: null,
7 verbose parsed url   slashes: null,
7 verbose parsed url   auth: null,
7 verbose parsed url   host: null,
7 verbose parsed url   port: null,
7 verbose parsed url   hostname: null,
7 verbose parsed url   hash: null,
7 verbose parsed url   search: null,
7 verbose parsed url   query: null,
7 verbose parsed url   pathname: 'figlet',
7 verbose parsed url   path: 'figlet',
7 verbose parsed url   href: 'figlet' }
8 silly lockFile b9071348-figlet figlet
9 verbose lock figlet C:\Users\ztr-hn\AppData\Roaming\npm-cache\b9071348-figlet.lock
10 silly lockFile b9071348-figlet figlet
11 silly lockFile b9071348-figlet figlet
12 verbose addNamed [ 'figlet', '' ]
13 verbose addNamed [ null, '*' ]
14 silly lockFile a40df646-figlet figlet@
15 verbose lock figlet@ C:\Users\ztr-hn\AppData\Roaming\npm-cache\a40df646-figlet.lock
16 silly addNameRange { name: 'figlet', range: '*', hasData: false }
17 verbose url raw figlet
18 verbose url resolving [ 'https://registry.npmjs.org/', './figlet' ]
19 verbose url resolved https://registry.npmjs.org/figlet
20 info trying registry request attempt 1 at 13:52:23
21 http GET https://registry.npmjs.org/figlet
22 error TypeError: Request path contains unescaped characters.
22 error     at new ClientRequest (_http_client.js:73:11)
22 error     at Object.exports.request (http.js:49:10)
22 error     at TunnelingAgent.exports.request (https.js:136:15)
22 error     at TunnelingAgent.createSocket (D:\nodejs\node_modules\npm\node_modules\request\node_modules\tunnel-agent\index.js:117:25)
22 error     at TunnelingAgent.createSecureSocket [as createSocket] (D:\nodejs\node_modules\npm\node_modules\request\node_modules\tunnel-agent\index.js:184:41)
22 error     at TunnelingAgent.addRequest (D:\nodejs\node_modules\npm\node_modules\request\node_modules\tunnel-agent\index.js:80:8)
22 error     at new ClientRequest (_http_client.js:154:16)
22 error     at Object.exports.request (http.js:49:10)
22 error     at Object.exports.request (https.js:136:15)
22 error     at Request.start (D:\nodejs\node_modules\npm\node_modules\request\request.js:583:30)
23 error If you need help, you may report this *entire* log,
23 error including the npm and node versions, at:
23 error     <http://github.com/npm/npm/issues>
24 error System Windows_NT 6.1.7601
25 error command "D:\\nodejs\\\\node.exe" "D:\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "install" "figlet"
26 error cwd C:\Users\ztr-hn
27 error node -v v0.12.0
28 error npm -v 1.4.9
29 verbose exit [ 1, true ]
```

How can I install a module using node?
Thank you.

**Comment by martinheidegger** [comment]
Thank you for writing it here. But this is a rather tough issue that I think has a better chance to be solved if you post your error at http://github.com/npm/npm/issues 

My educated guess would be that it has to do with your network settings (are you using a proxy or firewall?) But I think people at npm have a better answer to you.

**Comment by martinheidegger** [comment]
Could you post it on npm and add a link here?

**Comment by notmatt** [comment]
@burdokz These two lines are a bit odd to me:
```
2 info using npm@1.4.9
3 info using node@v0.12.0
```
Since node v0.12.0 shipped with npm 2.5.0 (or 2.5.1? anyway, much more recent than 1.4.9). Out of curiosity, how did you install node? Did you have an older install previously, perhaps npm wasn't upgraded correctly?

As a first step, you might try `npm update -g npm`. Hopefully that fixes things, but either way I agree with @martinheidegger that the team [at npm](https://github.com/npm/npm/issues) might be interested in hearing about your issues, at least so they can help anyone else who ends up there.

Let us know if this works?

**Comment by burdokz** [comment]
Hi martin and notmatt

Thank you for your response.
I had to install node without admin permission and followed this tutorial: http://abdelraoof.com/blog/2014/11/11/install-nodejs-without-admin-rights/

The problem was is that I messed up some command line while setting up the proxy and accidentaly changed the registry path to my own proxy path, that's what caused this whole problem for me. 
It took me a while to realize my error and after setting up (correctly now) the proxies and the registry it worked fine.

Thank you for your attention.
Hugo Nakamura
